packages/dicomImageLoader/src/imageLoader/wadors/extractMultipart.ts

```diff
@@ -0,0 +1,92 @@
+import findIndexOfString from './findIndexOfString';
+
+export function findBoundary(header: string[]): string {
+  for (let i = 0; i < header.length; i++) {
+    if (header[i].substr(0, 2) === '--') {
+      return header[i];
+    }
+  }
+}
+
+export function findContentType(header: string[]): string {
+  for (let i = 0; i < header.length; i++) {
+    if (header[i].substr(0, 13) === 'Content-Type:') {
+      return header[i].substr(13).trim();
+    }
+  }
+}
+
+export function uint8ArrayToString(data, offset, length) {
+  offset = offset || 0;
+  length = length || data.length - offset;
+  let str = '';
+
+  for (let i = offset; i < offset + length; i++) {
+    str += String.fromCharCode(data[i]);
+  }
+
+  return str;
+}
+
+export default function extractMultipart(
+  contentType: string,
+  imageFrameAsArrayBuffer,
+  progressiveContent?,
+  isPartial = false
+) {
+  // request succeeded, Parse the multi-part mime response
+  const response = new Uint8Array(imageFrameAsArrayBuffer);
+
+  if (contentType.indexOf('multipart') === -1) {
+    return {
+      contentType,
+      complete: !isPartial,
+      pixelData: response,
+    };
+  }
+
+  let { tokenIndex, responseHeaders, boundary, multipartContentType } =
+    progressiveContent || {};
```

Comment:
these are more `options` to me than anythign related to `progressiveContent`, you can even put the `isPartial` in the `options`